urlac47192b66_ggrandes_bouncer_tgz-pJ8-org_javastack_bouncer_ClusterTestJ8/src/org/javastack/bouncer/benchmark/BenchmarkThroughput.java

```diff
@@ -0,0 +1,156 @@
+package org.javastack.bouncer.benchmark;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Generate random text [0-9a-zA-Z] and consume InputStream
+ */
+public class BenchmarkThroughput {
+	private static final int CHARGEN_TCP = 19;
+	private static final char[] chars = "0123456789abcdefefghijklmnopqrstuvwxyzABCDEFEFGHIJKLMNOPQRSTUVWXYZ"
+			.toCharArray();
+	private static final ExecutorService threadPool = Executors.newCachedThreadPool();
+	private static final int THROUGHTPUT_BYTES = 512 * 1024 * 1024; // 512MB
+
+	public static void main(final String[] args) throws Exception {
+		final boolean client = Boolean.getBoolean("client");
+		if (client) {
+			Socket sock = args.length < 2 ? new Socket("127.0.0.1", CHARGEN_TCP) : //
+					new Socket(args[0], Integer.parseInt(args[1]));
+			System.out.println("Connected to: " + sock);
+			new SocketSync(sock).run();
+			threadPool.shutdown();
+		} else {
+			ServerSocket listen = new ServerSocket(args.length < 1 ? CHARGEN_TCP : Integer.parseInt(args[0]));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *listen*. Execution paths that do not contain closure statements were detected. Ensure *listen* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).